src/middlewared/middlewared/etc_files/local/collectd.conf

```diff
@@ -33,6 +33,9 @@
 		cpu_plugin_options = ''
 		aggregation_plugin_cpu_type = 'cpu'
 
+	ups_config = middleware.call_sync('ups.config')
+	ups_status = middleware.call_sync('service.started', 'ups')
```

Comment:
Why does it depend on whether ups is started or not?

Reply:
So if the service is not running, we won't have anything running on the ups port and when collectd tries to connect on the ups port, that will fail

Reply:
So it crashes collectd or spams to syslog or what?

Reply:
So it does not crash collectd, just logs that it is unable to connect with the port specified by ups config

Reply:
If that is acceptable, I can remove the conditions and we can write the ups configuration anyways then ? ( but as we are restarting collectd on each ups service action, I thought that it might be good that we add this check as well in the config ). Please let me know what you think best and we can move with that. Thank you

Reply:
What ensures that, when collectd is started on boot, if UPS service is desired to start, this condition is always true?

Reply:
That case is indeed left out. On boot, etc plugin has a higher priority in the rcorder so ups service won't be running when this condition is evaluated. I think we can either drop the check or introduce an or condition which makes this check true if ups service is set to start on boot. Thoughts ?

Reply:
> introduce an or condition which makes this check true if ups service is set to start on boot.

I think this, otherwise it's too complicated. No other service config generation is dependent on whether another service is started.
Also we still need to prevent collectd writing "can't connect to UPS" to syslog on startup if it is still started before ups service.

Reply:
I think that might get complicated. There are three scenarios
1) Ups config is valid. Boot is set to true for ups service
2) Ups config is invalid. Boot is set to true for ups service
3) Ups config is valid but perhaps ups daemon could not connect to ups and the driver exits for some reason.

So in order to secure point no 2, if we disable collectd writing can't connect to UPS, we can potentially miss point no 3 which is a valid case and should be logged I think.

Point no 1 if everything is good cannot happen because ups has a higher priority in the rcorder then collectd.

Should we move on with preventing the error message ? I don't think we can easily distinguish between case 2 and 3 ?

Reply:
> > introduce an or condition which makes this check true if ups service is set to start on boot.
> 
> I think this, otherwise it's too complicated. No other service config generation is dependent on whether another service is started.

Perfect, I'll move ahead with this then